### PR TITLE
Added commands for authorization in security realms

### PR DESCRIPTION
**Pull request by olukas**
New tests locally pass with all profiles (wildfly10, wildfly9, wildfly8, as7).

### CODE REVIEW

File: commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddLdapAuthorization.java [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+import org.wildfly.extras.creaper.core.offline.OfflineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Batch;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+/**
+ * Add ldap authorization to security realm.
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddLdapAuthorization.java]: LDAP

File: commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddLdapAuthorization.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+                .build());
+    }
+
+    private void addCache(Batch batch, Address address, LdapCache cache) {
+        if (cache != null) {
+            String cacheType = null;
+            if (cache.getByAccessTime()) {
+                cacheType = "by-access-time";
+            }
+            if (cache.getBySearchTime()) {
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddLdapAuthorization.java]: `else if`?

File: commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AdvancedFilter.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        return recursive;
+    }
+
+    public String getUserDnAttribute() {
+        return userDnAttribute;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AdvancedFilter.java]: Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.

File: commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/GroupToPrincipal.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        return recursive;
+    }
+
+    public String getSearchBy() {
+        return searchBy;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/GroupToPrincipal.java]: Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.

File: commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/PrincipalToGroup.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        return preferOriginalConnection;
+    }
+
+    public Boolean getSkipMissingGroups() {
+        return skipMissingGroups;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/PrincipalToGroup.java]: Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.

File: commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/UsernameFilter.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        return recursive;
+    }
+
+    public String getUserDnAttribute() {
+        return userDnAttribute;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/UsernameFilter.java]: Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.

File: commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/UsernameIsDn.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        this.cache = builder.cache;
+    }
+
+    public Boolean getForce() {
+        return force;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```
Comment (Ladicek) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/UsernameIsDn.java]: Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.

**Comment by Ladicek** [comment]
Few comments, otherwise LGTM. You're killing it with the tests! :-)

**Comment by olukas** [comment]
Issues fixed. Once you will confirm that current version is fine then it will be rebased and squashed.

**Comment by Ladicek** [comment]
LGTM. Please squash and I'll merge.

BTW, if you don't really need the offline variants (I can see that they can be quite painful to maintain), you don't have to provide them.

**Comment by olukas** [comment]
Done.

**Pull request merged**